src/Sleep.js

```diff
@@ -1,38 +1,25 @@
-class Sleep {
+import StatsParent from '../src/StatsParent'
+class Sleep extends StatsParent{
   constructor(sleepData, userID) {
+    super();
     this.sleepData = sleepData;
     this.userID = userID;
-  }
-
-  findUser() {
-    return this.sleepData.filter(user => user.userID === this.userID);
-  }
-
-  returnWeekOfData(week, userData) {
-    return [...userData].splice((-7 * week), 7);
-  }
-
-  returnWeek(week) {
-    var specificUser = this.findUser()
-    return [...specificUser].splice(-7 * week, 7).map(day => day.date);
+    this.finder = this.findUser(this.sleepData,this.userID)
   }
 
   returnAvgSleepData(relevantProperty) {
-      let specificUser = this.findUser();
-      return Number((specificUser.reduce((totalHours, day) => {
+      return Number((this.finder.reduce((totalHours, day) => {
         totalHours += day[relevantProperty];
         return totalHours;
-      }, 0) / specificUser.length).toFixed(2));
+      }, 0) / this.finder.length).toFixed(2));
     }
 
   returnSleepData(date, relevantProperty) {
```

Comment:
This is the same as 

returnUserDataForDay(userData, id, date, relevantProperty) {
    let specificUser = this.findUser(userData, id);
    return specificUser.find(day => day.date === date)[relevantProperty];
  }

in StatsParent so you should be able to take it out.

Reply:
I will take care of these 3.

Reply:
Cool, I made an issue and assigned it to you https://github.com/spaceplesiosaur/refactor-tractor/issues/59